src/json/builder.cr

```diff
@@ -94,35 +94,55 @@ class JSON::Builder
   # This method can also be used to write the name of an object field.
   def string(value)
     string = value.to_s
+
     scalar(string: true) do
       io << '"'
-      string.each_char do |char|
-        case char
+
+      pos = 0
```

Comment:
Please use a more descriptive name like `last_start_pos`.

Reply:
too long... maybe `start` or `start_pos`... theres only one pos to track, why last? theres no first. theres no middle. algorithm is simple.